### DISCUSSION TITLE
save files to specific path? YYYYMMDD/HH/Generation#

### DISCUSSION THREAD

**Discussion by Sacred-Reputation** [vladmandic/sdnext#1859] [General]
I'd like each generation to be saved into it's own subfolder under the date, then ONLY the hour, then the generation number..can't figure out how to get it to work

C:\outputs\YYMMDD\HH\Generation#

So if I ran a batch of 3 today at 3:15pm, it'd result in the following:

so it'd be C:\outputs\2023-07-28\15\01.png
                C:\outputs\2023-07-28\15\02.png
                C:\outputs\2023-07-28\15\03.png
                
and then another batch of 3 at 9:40pm they'd be:

                C:\outputs\2023-07-28\21\01.png
                C:\outputs\2023-07-28\21\02.png
                C:\outputs\2023-07-28\21\03.png                

any help?

**Comment by Tillerz** [comment]
Looking at the variables you can use there that doesn't exactly work.
You can only use [date] or [datetime] for the folder name, no single hour available right now.
And [generation_number] is only available for the filename, not the folder.

So all what you could to is

```
Images filename pattern: [seq]
Directory name pattern: [datetime]
```

to get close to what you want.

If you want to separate the stuff you do during a day, you might try

Directory name pattern: `[date]/[prompt_hash]` or `[date]_[prompt_hash]`

assuming that your prompt changes for those runs.


Looking at modules/images.py, you could add the hour only at least to the filename, can't find where you could do it for the folder:

```
class FilenameGenerator:
    replacements = {
        'batch_number': lambda self: NOTHING_AND_SKIP_PREVIOUS_TEXT if self.p.batch_size == 1 else self.p.batch_index + 1,
        'cfg': lambda self: self.p and self.p.cfg_scale,
        'clip_skip': lambda self: self.p and self.p.clip_skip,
        'date': lambda self: datetime.datetime.now().strftime('%Y-%m-%d'),
```

You could try adding this:

```
        'hour': lambda self: datetime.datetime.now().strftime('%H'),
```
